### ISSUE TITLE
Crop fills with white pixel when using DWAA EXR compression.

### ISSUE THREAD

**Issue by tokejepsen** [ImageMagick/ImageMagick#1760]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of ImageMagick
- [x] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

When converting to the DWAA compression for EXRs, any cropped pixels get filled with white.

### Steps to Reproduce

Convert any cropped EXR to a DWAA compression with the following command:

```
magick convert input.exr -compress DWAA output.exr
```

I have attached an image which contains a cropped roto shape (from Nuke), and an image convert with the above command to DWAA compression.

[exr_compression.zip](https://github.com/ImageMagick/ImageMagick/files/3738925/exr_compression.zip)

### System Configuration

- ImageMagick version: 

```
Version: ImageMagick 7.0.8-68 Q16 x64 2019-10-05 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2018 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Visual C++: 180040629
Features: Cipher DPC Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg lcms lqr lzma openexr pangocairo png ps raw rsvg tiff webp xml zlib
```

- Environment (Operating system, version and so on):

Windows 10

- Additional information:

**Comment by fmw42** [comment]
Yes, IM 6.9.10.68 and 7.0.8.68. But I listed above from IM 6. I see now that if I use IM 7 and list the compressions, it does list DWAA. Sorry about the confusion.

**Comment by tokejepsen** [comment]
Looks like its not just the DWAA compression. Converting any EXR that has a crop, ImageMagick will fill the cropped pixels with white.

**Comment by fmw42** [comment]
If you crop, try adding +repage after the crop. Does that help?

**Comment by tokejepsen** [comment]
> If you crop, try adding +repage after the crop to remove the virtual canvas geometry. Does that help?

Do you mean `magick convert +repage C:\Users\admin\Desktop\zip_compression.exr C:\Users\admin\Desktop\zip_compression.jpeg`?
That still has the white pixel fill.

**Comment by fmw42** [comment]
Remove the `convert` and put the +repage after the crop. 

`magick image.exr -crop WxH+X+Y +repage cropped_image.png`

+repage should not be needed for JPG output, which does not support virtual canvas. So if your output is JPG and you have cropped your image, the cropped out area should not show.

Post a link to an example exr image and the exact command line you are using.

**Comment by tokejepsen** [comment]
> Perhaps I misunderstand your issue. Please clarify where the crop is done. Your command shows no cropping!

Yeah, I may not have been very clear. The exr is already cropped from Nuke with its autocrop feature. Here is an EXR cropped from Nuke; 
[cropped.zip](https://github.com/ImageMagick/ImageMagick/files/3752656/cropped.zip)

Using this command results in the cropped pixel to fill with white;

```
magick cropped.exr cropped.jpeg
```

Resulting output:
![result](https://user-images.githubusercontent.com/1860085/67238685-7f77f700-f445-11e9-95a4-65c318c54343.jpeg)

Expected output:
![expected](https://user-images.githubusercontent.com/1860085/67238702-856dd800-f445-11e9-8afb-cd398c0b72cf.jpeg)

**Comment by fmw42** [comment]
Yes, Imagemagick is getting the wrong arguments for the size for the EXR image.

Here is what I get from EXIFTOOL:

`exiftool -s -ee -g1 -u -n -D cropped.exr
---- ExifTool ----
    - ExifToolVersion                 : 10.51
---- System ----
    - FileName                        : cropped.exr
    - Directory                       : .
    - FileSize                        : 50893
    - FileModifyDate                  : 2019:10:17 12:11:10-07:00
    - FileAccessDate                  : 2019:10:21 13:07:06-07:00
    - FileInodeChangeDate             : 2019:10:21 13:06:13-07:00
    - FilePermissions                 : 755
---- File ----
    - FileType                        : EXR
    - FileTypeExtension               : EXR
    - MIMEType                        : image/x-exr
    - ImageWidth                      : 695
    - ImageHeight                     : 562
---- OpenEXR ----
    - EXRVersion                      : 2
    - Layout                          : 0
    - Channels                        : B half 1 1, G half 1 1, R half 1 1
    - Compression                     : 2
    - DataWindow                      : 650 24 1344 585
    - DisplayWindow                   : 0 0 1919 1079
    - LineOrder                       : 0
    - Nukefull_layer_names            : 0
    - Nukenode_hash                   : 48ff295dda5c5066
    - Nukeversion                     : 11.0v4
    - PixelAspectRatio                : 1
    - ScreenWindowCenter              : 0 0
    - ScreenWindowWidth               : 1
    - Type                            : scanlineimage
    - Version                         : 1
---- Composite ----
    - ImageSize                       : 695x562
    - Megapixels                      : 0.39059
`

Note the ImageSize parameter near the bottom as 695x562.  But Imagemagick is using:

`cropped.exr
Image: cropped.exr
  Format: EXR (High Dynamic-range (HDR))
  Class: DirectClass
  Geometry: 1920x1080+0+0
`
1920x1080  which comes from what is shown in EXIFTOOL as:

`- DisplayWindow                   : 0 0 1919 1079`

So I believe there might be a bug that the Imagemagick developers will need to look into.

**Comment by dlemstra** [comment]
You can control the background color with `-background`:

`convert -background black input.exr output.jpg`

**Comment by tokejepsen** [comment]
Thanks @dlemstra! That'll work as a workaround atm.

To be specific we need to specify the background and alpha:

`magick convert -background black -alpha copy input.exr output.png`

It does remove the built-in cropping when converting between exrs, so not sure whether Image Magick want to look into this:

`magick convert -background black -alpha copy input.exr output.exr`

Personally I can live with the workaround solution.

**Comment by dlemstra** [comment]
What is the result that you are expecting? Do you want the cropped image instead?

**Comment by tokejepsen** [comment]
> What is the result that you are expecting? Do you want the cropped image instead?

I would expect the exrs to match when converting between different compressions.
So if we compress the `cropped.exr` with:

`magick convert -background black -alpha copy cropped.exr -compress DWAA result.exr`

```
C:\Users\admin>"C:\Users\admin\Desktop\exiftool-11.71\exiftool(-k).exe" -s -ee -g1 -u -n -D C:\Users\admin\Desktop\cropped.exr
---- ExifTool ----
    - ExifToolVersion                 : 11.71
---- System ----
    - FileName                        : cropped.exr
    - Directory                       : C:/Users/admin/Desktop
    - FileSize                        : 37117
    - FileModifyDate                  : 2019:10:22 07:37:23+01:00
    - FileAccessDate                  : 2019:10:22 07:37:23+01:00
    - FileCreateDate                  : 2019:10:21 20:56:23+01:00
    - FilePermissions                 : 666
---- File ----
    - FileType                        : EXR
    - FileTypeExtension               : EXR
    - MIMEType                        : image/x-exr
    - ImageWidth                      : 522
    - ImageHeight                     : 394
---- OpenEXR ----
    - EXRVersion                      : 2
    - Layout                          : 0
    - Channels                        : A half 1 1, B half 1 1, G half 1 1, R half 1 1
    - Compression                     : 2
    - DataWindow                      : 660 190 1181 583
    - DisplayWindow                   : 0 0 1919 1079
    - LineOrder                       : 0
    - Nukefull_layer_names            : 0
    - Nukenode_hash                   : c22743bb50eb6fb8
    - Nukeversion                     : 11.0v4
    - PixelAspectRatio                : 1
    - ScreenWindowCenter              : 0 0
    - ScreenWindowWidth               : 1
    - Type                            : scanlineimage
    - Version                         : 1
---- Composite ----
    - ImageSize                       : 522 394
    - Megapixels                      : 0.205668
```

```
C:\Users\admin>"C:\Users\admin\Desktop\exiftool-11.71\exiftool(-k).exe" -s -ee -g1 -u -n -D C:\Users\admin\Desktop\result.exr
---- ExifTool ----
    - ExifToolVersion                 : 11.71
---- System ----
    - FileName                        : result.exr
    - Directory                       : C:/Users/admin/Desktop
    - FileSize                        : 45378
    - FileModifyDate                  : 2019:10:22 07:42:41+01:00
    - FileAccessDate                  : 2019:10:22 07:42:41+01:00
    - FileCreateDate                  : 2019:10:22 07:34:57+01:00
    - FilePermissions                 : 666
---- File ----
    - FileType                        : EXR
    - FileTypeExtension               : EXR
    - MIMEType                        : image/x-exr
    - ImageWidth                      : 1920
    - ImageHeight                     : 1080
---- OpenEXR ----
    - EXRVersion                      : 2
    - Layout                          : 0
    - Channels                        : A half 1 1, B half 1 1, G half 1 1, R half 1 1
    - Compression                     : 8
    - DataWindow                      : 0 0 1919 1079
    - DisplayWindow                   : 0 0 1919 1079
    - LineOrder                       : 0
    - PixelAspectRatio                : 1
    - ScreenWindowCenter              : 0 0
    - ScreenWindowWidth               : 1
---- Composite ----
    - ImageSize                       : 1920 1080
    - Megapixels                      : 2.0736
```

We end up with a `result.exr` that has the wrong `DataWindow`, `660 190 1181 583` compared to `0 0 1919 1079`.

**Comment by dlemstra** [comment]
It took me a while but I finally understand that you would like to preserve the DataWindow. ImageMagick currently does not preserve this. Would it help you if it would be possible to only read the DataWindow and get an image that is only the cropped area.

**Comment by tokejepsen** [comment]
> Would it help you if it would be possible to only read the DataWindow and get an image that is only the cropped area?

I would expect to preserve the DataWindow.

BUT with the solution of flooding the background with black transparent pixels works for me, so consider this issue non-priority and just something to flag for the future if someone else are having the same issue.

**Comment by tokejepsen** [comment]
For anyone following this or finding it later, you dont need `-alpha copy` since the cropped pixels are empty the correct command is:

```
magick convert -background "rgba(0, 0, 0, 0)" cropped.exr result.exr
```

**Comment by fmw42** [comment]
In Imagemagick 7, use magick, not magick convert. The latter will give Imagemagick 6 behavior.

**Comment by lola-post** [comment]
We would like to +1 this behaviour as incorrect.

EXR's have a datawindow (autocrop) that needs to be preserved.
We would not expect a cropped image, but rather an image with the original resolution, with a datawindow inside it that could be bigger or smaller than the resolution of the image.

In vfx we use this for various reasons, examples:
- overscan
- autocropped cgo render (where an object may only fill a few pixels so the exr can be very small with the same resolution) In our applications, this function is called autocrop.
